### PR TITLE
tests: Fix functional tests to run with Java1.7.

### PR DESCRIPTION
**Pull request by harshavardhana**
java1.7 defaults to using TLSv1.1 but Minio server
only support TLSv1.2 by default.

This PR addresses a problem which surfaced recently
while using Java 7 with Minio server.

### CODE REVIEW

**Comment by harshavardhana** [comment]
While this works @balamurugana , i need to make this selectable i.e need to change it use cleartext with HTTP v/s HTTPS. Until then :-1:

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -927,6 +937,23 @@ private HttpResponse execute(Method method, String region, String bucketName, St
...
+    }
+
+    SSLSocketFactory socketFac = context.getSocketFactory();
+
+    ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+        .tlsVersions(TlsVersion.TLS_1_2)
+        .build();
+
+    this.httpClient.setConnectionSpecs(Collections.singletonList(spec));
+    this.httpClient.setSslSocketFactory(socketFac);
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: * You could make this as a separate public method `public void enableTLSv12()`.  This make a choice to consumer to choose what they want than forcing to TLSv1.2.
* Its not required for each `execute()`
* you can directly like `this.httpClient.setSslSocketFactory(context.getSocketFactory());`

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -927,6 +937,23 @@ private HttpResponse execute(Method method, String region, String bucketName, St
       this.traceStream.println(headers);
     }
 
+    SSLContext context = SSLContext.getInstance("TLSv1.2");
+
+    try {
+      context.init(null, null, null);
+    } catch (KeyManagementException e) {
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: You can raise `KeyManagementException` itself than eating with `InternalException`

File: functional/FunctionalTest.java [hunk 3]
```diff
@@ -649,7 +654,24 @@ public static void presignedGetObject_test1() throws Exception {
...
+    }
+
+    SSLSocketFactory socketFac = context.getSocketFactory();
+    ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+        .tlsVersions(TlsVersion.TLS_1_2)
+        .build();
+
+    transport.setConnectionSpecs(Collections.singletonList(spec));
+    transport.setSslSocketFactory(socketFac);
+
```
Comment (balamurugana) [inline functional/FunctionalTest.java]: You could have one `transport` as global where TLSv1.2 is enabled only for `play.minio.io` in `main()`.  This way repetition code is removed.

File: functional/FunctionalTest.java [hunk 4]
```diff
@@ -649,7 +654,24 @@ public static void presignedGetObject_test1() throws Exception {
         .url(HttpUrl.parse(urlString))
         .method("GET", null)
         .build();
+
     OkHttpClient transport = new OkHttpClient();
+    SSLContext context = SSLContext.getInstance("TLSv1.2");
+
+    try {
+      context.init(null, null, null);
+    } catch (KeyManagementException e) {
```
Comment (balamurugana) [inline functional/FunctionalTest.java]: You can throw `KeyManagementException` itself.

File: functional/FunctionalTest.java [hunk 5]
```diff
@@ -899,7 +921,24 @@ public static void presignedGetObject_test2() throws Exception {
...
+    }
+
+    SSLSocketFactory socketFac = context.getSocketFactory();
+    ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+        .tlsVersions(TlsVersion.TLS_1_2)
+        .build();
+
+    transport.setConnectionSpecs(Collections.singletonList(spec));
+    transport.setSslSocketFactory(socketFac);
+
```
Comment (poornas) [inline functional/FunctionalTest.java]: This code block can be refactored into a separate method, instead of repeating for each presigned url test cases.

**Pull request closed without merging**